### DISCUSSION TITLE
Intercept Traffic vom IOT Device

### DISCUSSION THREAD

**Discussion by DeepflashX** [mitmproxy/mitmproxy#6489] [Intercept Everything!]
Hello altogether,

this is my first attempt to use MITMProxy and therefor I would gladly appreciate some help in how the steps are.
So I have an IOT device which has wifi-chip for which I read out the Firmware and found the SSL-certificate and can replace that with my own.
Then I want to route the traffic to MITMproxy, decrypt and encrypt again with original certifcate.

The device connects via WIFI to my Fritzbox and targets a mqtts server externally.
I would reroute that traffic on my Fritzbox to the machine on which mitmproxy is running and then get it back on original receiver.
That should show me what kind of information from the mqtts server is being received, right?
Now from the theory to the doing: how do I need to setup MITMProxy? I am kinda confused with all the modes, would that be transparent Proxy Variant 3?
Is traffic on MQTTS working at all because I mainly read about HTTP-Proxy?

I am open for any tips and support.

Best regards,
Matthias

**Comment by avabob** [comment]
Hi! I'm curious if you had any luck with this method. I've also been running into some trouble capturing MQTTS traffic. If you were successful, would you mind sharing what you did?
As shown in the image, the MQTTS traffic appears to have been parsed as HTTPS.
<img width="1056" height="289" alt="image" src="https://github.com/user-attachments/assets/cebe2ca3-d402-4c53-8906-ffcd391cd201" />
<img width="1648" height="576" alt="image" src="https://github.com/user-attachments/assets/c07df242-a5f3-4247-a585-d4378f2b656b" />
